File: templates/python/fr/text/code/beeperLine.py

# Fichier: JetonLigne.py
# ------------------------------
# Utilise un boucle while pour placer une ligne de jetons .
# Ce programme fonctionne pour un monde de toute taille.
from karel.stanfordkarel import *

# programme commence à main
def main():
   # se répète jusqu'à ce que Karel fasse face à un mur
   while l'avant_est_clair():
      # place un jeton sur la case actuelle
      laisser_jeton()
      # avancer au carré suivant
      avancer()
   # résout le bogue de clôture
   laisser_jeton()

File: templates/java/fr/text/code/beeperLine.java
/*
 * Fichier: LigneJeton.java
 * ------------------------
 * Utilise une boucle "while" pour placer une ligne d
 * e jetons . Ce programme fonctionne pour un monde d
 * e toute taille.
 */
import stanford.karel.*;
public class LigneJeton extends Karel {
   public void run() {
      /* répète jusqu'à ce que Karel soit face à un mur */
      while(devantDegage()) {
         /* placer un jeton sur le carré actuel */ 
         laisserJeton();
         /* avancer à la prochaine case */
         avancer();
      }
      /* résout le bogue de clôture */
      laisserJeton();
   }
}